front/src/applications/operationalStudies/components/SimulationResults/SimulationResultsMap.tsx

```diff
@@ -85,6 +84,19 @@ interface MapProps {
   setExtViewport: (viewport: Viewport) => void;
 }
 
+interface InterpoledTrainIds {
+  name: string;
+  id: number;
+}
+
+interface InterpolatedTrainValues {
+  head_positions?: PositionSpeedTime;
+  tail_positions?: PositionSpeedTime;
+  speeds?: PositionSpeedTime;
+}
+
+type InterpoledTrain = InterpoledTrainIds & InterpolatedTrainValues;
```

Comment:
Why have you created 3 separated interface/types here ?
```suggestion
type InterpoledTrain {
  name: string;
  id: number;
  head_positions?: PositionSpeedTime;
  tail_positions?: PositionSpeedTime;
  speeds?: PositionSpeedTime;
}

```

Reply:
ok it's less verbose